nupic/research/frameworks/pytorch/imagenet/imagenet_experiment.py

```diff
@@ -193,6 +193,14 @@ def setup_experiment(self, config):
                               params_sparse, nonzero_params_sparse2,
                               float(nonzero_params_sparse2) / params_sparse)
 
+        modify_init_hooks = config.get("modify_init_hooks", [])
+        if not isinstance(modify_init_hooks, list):
+            modify_init_hooks = [modify_init_hooks]
+        if modify_init_hooks:
+            for modify_init in modify_init_hooks:
+                modified_model = modify_init(self.model, config=config)
```

Comment:
avoid passing the whole config to the callbacks/hooks. 
You should create a separate entry for each `modify_init_hooks` defining its arguments, something similar to the LR scheduler args. Otherwise it will be hard to differentiate what parameter belongs to what hook. Ideally, these hooks should not use a dict as a single argument but be a function with named arguments and default values

Reply:
Fair point